Plugins/BundleEditorPlugin/BundleEditor.cs

```diff
@@ -11,9 +12,286 @@
 using System.Windows.Input;
 using System.Windows.Media;
 using FrostySdk.Managers.Entries;
+using FrostySdk;
 
 namespace BundleEditPlugin
 {
+    #region RemoveFromBundleExtension
+    public class RemoveMeshExtension : RemoveFromBundleExtension
+    {
+        public override string AssetType => "MeshAsset";
+        public override void RemoveFromBundle(EbxAssetEntry entry, BundleEntry bentry)
+        {
+            base.RemoveFromBundle(entry, bentry);
+
+            EbxAsset asset = App.AssetManager.GetEbx(entry);
+            dynamic meshAsset = asset.RootObject;
+
+            //Add res to BUNDLES AND LINK
+            ResAssetEntry resEntry = App.AssetManager.GetResEntry(meshAsset.MeshSetResource);
+            resEntry.AddedBundles.Remove(App.AssetManager.GetBundleId(bentry));
+            
+            entry.LinkAsset(resEntry);
+
+            MeshSet meshSetRes = App.AssetManager.GetResAs<MeshSet>(resEntry);
+
+            //Double check if there are any LODs the mesh, if there are, bundle and link them
+            // J-Lyt | If chunk is in SuperBundle, do not add to bundle.
+            if (meshSetRes.Lods.Count > 0)
+            {
+                foreach (MeshSetLod lod in meshSetRes.Lods)
+                {
+                    ChunkAssetEntry chunkEntry = App.AssetManager.GetChunkEntry(lod.ChunkId);
+
+                    if (lod.ChunkId != Guid.Empty && chunkEntry.SuperBundles.Count == 0)
+                    {
+                        chunkEntry.AddedBundles.Remove(App.AssetManager.GetBundleId(bentry));
+                        resEntry.LinkAsset(chunkEntry);
+                    }
+                }
+            }
```

Comment:
Would it be better to do the guid null check before :

```suggestion
            if (meshSetRes.Lods.Count > 0)
            {
                foreach (MeshSetLod lod in meshSetRes.Lods)
                {
                    if (lod.ChunkId != Guid.Empty)
                    {
                        ChunkAssetEntry chunkEntry = App.AssetManager.GetChunkEntry(lod.ChunkId);

                        if (chunkEntry != null && chunkEntry.SuperBundles.Count == 0)
                        {
                            chunkEntry.AddedBundles.Remove(App.AssetManager.GetBundleId(bentry));
                            resEntry.LinkAsset(chunkEntry);
                        }
                    }
                }
            }
```

Reply:
Yeah that makes sense. I've added your changes and also added the MeshSetPlugin to the projectreferences as I forgot to add that to the commit. | https://github.com/wavebend/FrostyToolsuite/pull/2/commits/61353dd864df321962857d704e61a416c038cc68